spec/fixtures/modules/test/lib/puppet/functions/frozen_function.rb

```diff
@@ -1,5 +1,5 @@
 Puppet::Functions.create_function(:frozen_function) do
   def frozen_function(value)
-    value.reverse!
+    value.frozen?
```

Comment:
Sure, but this is a unit test to check the arguments are frozen rather than a test of `reverse!` or other methods that actually modify an object. Checking if the object's `#frozen?` is equivalent to calling a method that raises an exception `if frozen?`.